src/components/dialogs/commons/custom-mui-dialog/custom-mui-dialog.tsx

```diff
@@ -29,27 +44,33 @@ const CustomMuiDialog = ({
     onValidationError,
     titleId,
     disabledSave,
-    removeOptional,
-    ...dialogProps
+    removeOptional = false,
+    onCancel,
+    children,
 }) => {
     const { handleSubmit } = formMethods;
 
-    const handleClose = (event) => {
+    const handleCancel = (event: React.MouseEvent) => {
+        onCancel && onCancel();
         onClose(event);
     };
 
-    const handleCancel = (event) => {
+    const handleClose = (event: React.MouseEvent, reason?: string) => {
+        if (reason === 'backdropClick' && onCancel) {
+            handleCancel(event);
```

Comment:
```suggestion
            onCancel(event);
```
Otherwise you call onClose 2 times